lib/open_telemetry/tracer.ex

```diff
@@ -0,0 +1,63 @@
+defmodule OpenTelemetry.Tracer do
+  @moduledoc false
+
+  defmacro __using__(tracer) do
+    case tracer do
+      [] ->
+        quote do
+          defp __get_tracer__() do
+            :opentelemetry.get_tracer()
+          end
+        end
+      _ ->
+        quote do
+          defp __get_tracer__() do
+            :opentelemetry.get_tracer(unquote(tracer))
+          end
+        end
+    end
+  end
+
+  defmacro start_span(name, opts \\ quote(do: %{}), do: block) do
```

Comment:
To not require an additional function.

Reply:
Would that really be so expensive to call? How would look the stack traces in both cases?

Also I think that `start_trace/{2,3}` with block would be really strange as it would start **and** end span. I think it should be named `with_span`.

Reply:
We should strive to add no overhead over what is necessary. Since Elixir has real macros, unlike Erlang, I'd, as a user, expect a macro for wrapping spans.

But true about the name. Maybe should be `with_span`.